common/ledger/blkstorage/block_serialization.go

```diff
@@ -26,21 +26,14 @@ type txindexInfo struct {
 }
 
 func serializeBlock(block *common.Block) ([]byte, *serializedBlockInfo, error) {
-	buf := proto.NewBuffer(nil)
-	var err error
+	var buf []byte
 	info := &serializedBlockInfo{}
 	info.blockHeader = block.Header
 	info.metadata = block.Metadata
-	if err = addHeaderBytes(block.Header, buf); err != nil {
-		return nil, nil, err
-	}
-	if info.txOffsets, err = addDataBytesAndConstructTxIndexInfo(block.Data, buf); err != nil {
-		return nil, nil, err
-	}
-	if err = addMetadataBytes(block.Metadata, buf); err != nil {
-		return nil, nil, err
-	}
-	return buf.Bytes(), info, nil
+	buf = addHeaderBytes(block.Header, buf)
+	info.txOffsets, buf = addDataBytesAndConstructTxIndexInfo(block.Data, buf)
+	buf = addMetadataBytes(block.Metadata, buf)
+	return buf, info, nil
```

Comment:
if function not going to return and error, I guess you can get rid of it in function signature also.

Reply:
fix